src/io/output_vtk.cc

```diff
@@ -224,7 +221,7 @@ void OutputVTK::fill_element_types_vector(std::vector< unsigned int >& data)
 
 
 
-void OutputVTK::write_vtk_data_ascii(OutputTime::OutputDataPtr output_data, VTKValueType type)
+void OutputVTK::write_vtk_data(OutputTime::OutputDataPtr output_data, VTKValueType type)
```

Comment:
Introduce VTKValueType into OutputDataBase and set it in constructors of derived classes using a helper function returning VTKValuteType specialized for the template parameters.

Then we can remove the parameter 'type' of this method and guarantee correct  VTK type of output data.